app/models/email_event.rb

```diff
@@ -10,15 +10,21 @@
 #  to           :string           not null
 #  created_at   :datetime         not null
 #  updated_at   :datetime         not null
+#  message_id   :string
 #
 class EmailEvent < ApplicationRecord
+  RETENTION_DURATION = 1.month
+
   enum status: {
     dispatched: 'dispatched',
     dispatch_error: 'dispatch_error'
   }
 
-  scope :dolist, -> { where(method: 'dolist') }
+  scope :dolist, -> { dolist_smtp.or(dolist_api) }
+  scope :dolist_smtp, -> { where(method: 'dolist_smtp') }
+  scope :dolist_api, -> { where(method: 'dolist_api') }
   scope :sendinblue, -> { where(method: 'sendinblue') }
+  scope :outdated, -> { where("created_at < ?", (Time.zone.now - RETENTION_DURATION)) }
```

Comment:
```suggestion
  scope :outdated, -> { where("created_at < ?", RETENTION_DURATION.ago) }
```

(ou un truc m'échappe ?)